src/code_805A120.c

```diff
@@ -0,0 +1,1474 @@
+extern void sub_80943A0(void*, s32);
+s32 RoundUpFixedPoint(s32 fixedPointNumber);
+extern s16 sub_8057600(struct Move*, u32);
+extern void sub_806CDD4(struct Entity *, u32, u32);
+extern bool8 sub_80705F0(struct Entity *pokemon, struct Position *pos);
+extern u8 sub_8044B28(void);
+extern u8 sub_803F428(struct Position *pos);
+extern void sub_804539C(struct Entity *, u32, u32);
+extern void sub_803E46C(u32);
+bool8 sub_80571F0(struct Entity * pokemon, struct Move *move);
+extern void sub_807EC28(u32);
+extern void sub_806F370(struct Entity *r0, struct Entity *r1, u32, u32, u8 *, u8, s32, u32, u32, u32);
+void sub_8075900(struct Entity *pokemon, u8 r1);
+extern void sub_804535C(struct Entity *, u32);
+extern void sub_804AC20(struct Position *);
+extern void sub_806A5B8(struct Entity *entity);
+extern void sub_80694C0(struct Entity *, s32, s32, u32);
+bool8 sub_80706A4(struct Entity *pokemon, struct Position *pos);
+extern void sub_807D148(struct Entity *pokemon, struct Entity *target, u32 r2, struct Position *r3);
+extern void SetMessageArgument(u8 *buffer, struct Entity *r1, u32);
+extern void SetMessageArgument_2(u8 *buffer, struct EntityInfo *r1, u32);
+void sub_80522F4(struct Entity *r0, struct Entity *r1, char r2[]);
+extern void sub_806A6E8(struct Entity *);
+extern u32 sub_8055640(struct Entity *, struct Entity *, struct Move *, u32, u32);
+extern void sub_807EA30(u32);
+extern bool8 sub_805727C(struct Entity *, struct Entity *, s16);
+extern void sub_806ABAC(struct Entity *, struct Entity *);
+extern u8 sub_807EAA0(u32, u32);
+extern void sub_8045C28(struct Item *, u8 , u8);
+extern u8 sub_8057308(struct Entity *, u32);
+extern void sub_805A7D4(struct Entity *, struct Entity *, struct Item *, struct Position *);
+extern void sub_8045394(struct Entity *, s32, s32);
+extern void sub_804652C(struct Entity *, struct Entity *, struct Item *, u32, struct Position *);
+extern void sub_806EAF4(struct Entity *, struct Entity *, u8, u32, u32, s32 *, u32, u16, u32);
+extern u8 *gUnknown_80FEB60[];
+extern u8 *gUnknown_80FE330[];
+extern u8 *gUnknown_80FE36C[];
+extern u8 *gPtrForecastPreventsTypeSwitchMessage[];
+
+struct unkStruct_80928C0
+{
+    u32 unk0;
+    s32 unk4;
+    u8 unk8;
+    u8 unk9;
+};
+
+void sub_80928C0(u8 *buffer, struct Move *move, struct unkStruct_80928C0 *param_3);
+extern u8 sub_806F4A4(struct Entity *, u32);
+extern void sub_807DF38(struct Entity *pokemon, struct Entity *target, struct Position *pos, u32, u8 moveType, s16);
+
+
+bool8 sub_8059424(struct Entity *pokemon, struct Entity *target, struct Move *move, u32 param_4)
+{
+  bool8 flag;
+  
+  flag = FALSE;
+  if (sub_8055640(pokemon,target,move,0x100,param_4) != 0) {
+    flag = TRUE;
+    if(sub_805727C(pokemon, target, gUnknown_80F4DF0))
+    {
+        CringeStatusTarget(pokemon, target, FALSE);
+    }
+  }
+  return flag;
+}
+
+bool8 sub_805946C(struct Entity * pokemon,struct Entity * target,struct Move * move,u32 param_4)
```

Comment:
This applies recoil damage for moves like Double-Edge.

Reply:
For all recoil?

Reply:
Will rename to `RecoilMoveAction`